tests/lean/interactive/hover.lean

```diff
@@ -11,3 +13,75 @@ example (n : Nat) : True := by
   | Nat.zero => _
   --^ textDocument/hover
   | n + 1 => _
+
+
+/-- My tactic -/
+macro "mytac" o:("only"?) e:term : tactic => `(exact $e)
+
+example : True := by
+  mytac only True.intro
+--^ textDocument/hover
+      --^ textDocument/hover
+           --^ textDocument/hover
+
+/-- My way better tactic -/
+macro_rules
+  | `(tactic| mytac $[only]? $e) => `(apply $e)
+
+example : True := by
+  mytac only True.intro
+--^ textDocument/hover
+
+/-- My ultimate tactic -/
+elab_rules : tactic
+  | `(tactic| mytac $[only]? $e) => `(tactic| refine $e) >>= Lean.Elab.Tactic.evalTactic
+
+example : True := by
+  mytac only True.intro
+--^ textDocument/hover
+
+
+/-- My notation -/
+macro "mynota" e:term : term => e
+
+#check mynota 1
+     --^ textDocument/hover
+
+/-- My way better notation -/
+macro_rules
+  | `(mynota $e) => `(2 * $e)
+
+#check mynota 1
+     --^ textDocument/hover
+
+-- macro_rules take precedence over elab_rules for term/command, so use new syntax
+syntax "mynota'" term : term
```

Comment:
What's the reason for preferring `macro_rules` in this case but using the `elab_rules` for tactics? Do you think it could get confusing for users?

Reply:
It could, but we should discuss that outside of this PR. /cc @leodemoura